service/history/shard/context.go

```diff
@@ -496,6 +496,7 @@ Create_Loop:
 			switch err.(type) {
 			case *shared.WorkflowExecutionAlreadyStartedError,
 				*persistence.WorkflowExecutionAlreadyStartedError,
+				*persistence.CurrentWorkflowConditionFailedError,
```

Comment:
If the range ID changes, it will throw this CurrentWorkflowConditionFailedError? If it is the case, why no update the range id?

Reply:
CurrentWorkflowConditionFailedError in StartWorkflow only happened when concurrent record messed up. RangeID changes will not lead to this error. That's my understanding based on reading of https://github.com/uber/cadence/blob/master/common/persistence/cassandra/cassandraPersistence.go#L1229 - L1242

Reply:
I think @yycptt makes a point. should we retry on this error

Reply:
this error will be retry by client; it should be non-retriable from history service point of view currently.